pkg/yamlmeta/type.go

```diff
@@ -45,22 +46,26 @@ type DocumentType struct {
 	ValueType Type // typically one of: MapType, ArrayType, ScalarType
 }
 type MapType struct {
-	Items []*MapItemType
+	Items       []*MapItemType
+	annotations template.NodeAnnotations
```

Comment:
im -1 on this. this violates layering that's been setup. yamlmeta package should NOT know anything about templating.

Reply:
Can you explain what do you mean by layering?

Reply:
we would like to maintain boundaries between various packages so that there is a clean layering going on in the "cake" (project). bottom layers of the cake should not depend on higher layers of the cake.

one way to "test this" is to imagine if to use particular package in isolation. eg my other project needs to parse yamls and extract metadata (comments). it doesnt need any concept of templating and i want to interpret comments in my own way (eg `# foo` is something special, everything else is plain comment).

the way that ytt cake is set up is something like this:

- yamlmeta - standalone package that knows how to parse YAML and produce YAML AST.
  - no dependency
- template - standalone generic starlark templating with a high level instructions interface for building structures
  - no dependency
- texttemplate/yamltemplate - language specific (YAML, text, may be JSON in future, etc)
  - yamltemplate -> depends on -> yamlmeta, and template
  - texttemplate -> depends on -> template

(more pkgs descriptions here: https://github.com/vmware-tanzu/carvel-ytt/blob/develop/docs/dev.md#source-code-structure).